packages/verification-vue-component/src/components/CertifactionVerification.vue

```diff
@@ -158,19 +158,31 @@ export default {
                 console.log(e)
             }
         },
-        async verifyItem(item, key) {
+        async verifyItem(item, key, updating = false) {
             const hash = await hashingService.hashFile(item.file)
             let verification = await this.certifactionEthVerifier.verify(hash)
 
-            Vue.set(this.verificationItems, key, { ...item, ...verification })
+            if (updating) {
+                if (this.itemHasUpdated(this.verificationItems[key], verification)) {
+                    Vue.set(this.verificationItems, key, { ...item, ...verification })
+                }
+            } else {
+                Vue.set(this.verificationItems, key, { ...item, ...verification })
+            }
 
             if (this.offchainVerifier) {
                 verification = await this.offchainVerification(verification)
             }
 
             verification.loaded = true
 
-            Vue.set(this.verificationItems, key, { ...item, ...verification })
+            if (updating) {
```

Comment:
At this point I would check if the item is in the desired state, if not use a timeout to self-call the function again. Pretty simple, no check if it's updating or has updated.

Reply:
Changed as discussed